static-assets/scripts/main.js

```diff
@@ -180,6 +180,19 @@
       };
 
       CrafterCMSNext.renderBackgroundUI();
+
+      const darkModeQuery = window.matchMedia('(prefers-color-scheme: dark)');
+      if (darkModeQuery.matches) {
+        $('body').addClass('dark-mode');
+      }
+
+      darkModeQuery.onchange = (e) => {
+        if (e.matches) {
+          $('body').addClass('dark-mode');
+        } else {
+          $('body').removeClass('dark-mode');
+        }
+      };
```

Comment:
Do we even need this? Whilst is based on OS/navigator preference, probably the CSS media query does the trick.

This way, we get rid of the **duplicated code** (👎🏻).